### ISSUE TITLE
How to wait for deployment to finish

### ISSUE THREAD

**Issue by dfishburn** [nickytd/kubernetes-logging-helm#9]
Does anyone know how to wait for the deployment to finish?

I have done this in the past with Elastic (ELK) using:
```
kubectl rollout status deployment/es-client-ek-cluster --namespace ${K8S_NAMESPACE}
```

I see these helm charts create 2 different deployments:
```
$ kubectl -n os-logging get deployments
NAME                        READY   UP-TO-DATE   AVAILABLE   AGE
ofd-client                  0/1     1            0           2s
ofd-opensearch-dashboards   0/1     1            0           2s
```

Both of these commands finish quite early, while (using k9s) I see pods in the namespace still being dropped and recreated.
```
$ kubectl -n os-logging  rollout status deployment/ofd-client
Waiting for deployment "ofd-client" rollout to finish: 1 old replicas are pending termination...
Waiting for deployment "ofd-client" rollout to finish: 1 old replicas are pending termination...
deployment "ofd-client" successfully rolled out

$ kubectl -n os-logging  rollout status deployment/ofd-opensearch-dashboards
deployment "ofd-opensearch-dashboards" successfully rolled out
```

**Comment by nickytd** [comment]
Hi 
can you please check the pod status?
And do you observe the same behaviour with the latest released chart version (3.2.8)

Thanks
Niki

**Comment by nickytd** [comment]
Hi, 

You can check the init container logs. These are used for a startup dependency management. Initialisation jobs typically wait the dependent components to be up and running. For example, kafka init containers wait for zookeepers to be reachable. Index-patterns jobs wait for opensearch-dashboards to be in healthy state and the latter waits for opeansearch stack to be fully initialised so it can authenticate itself and uploads the "index-patterns", "searches" and "visualisations"
Don't use --wait as that will break the dependency checks. 
```helm install ... --timeout 10m```usually is sufficient.  The timeout here is needed to give sufficient time for the statefulsets to be initialised. This can take time due to the sequential startup order and the time volumes are created and attached. 

Take a look at [nickytd/kubernetes-logging](https://github.com/nickytd/kubernetes-logging) for a setup example.

Thanks
Niki
